src/transformers/modeling_tf_bart.py

```diff
@@ -458,16 +467,19 @@ def call(
         )
         x = tf.nn.dropout(x, rate=self.dropout if training else 0)
         x = residual + x
-
-        x = self.encoder_attn_layer_norm(x)
-
+        if not self.normalize_before:
+            x = self.encoder_attn_layer_norm(x)
+        # Fully Connected
         residual = x
+        if self.normalize_before:
+            x = self.final_layer_norm(x)
         x = self.activation_fn(self.fc1(x))
         x = tf.nn.dropout(x, rate=self.activation_dropout if training else 0)
         x = self.fc2(x)
         x = tf.nn.dropout(x, rate=self.dropout if training else 0)
```

Comment:
Had not seen last review, why use `tf.nn.dropout`  and `tf.keras.layers.Dropout` and not stick with a single one?

Reply:
The reason was pytorch consistency, but I will fix both.